pygeo/constraints/thicknessConstraint.py

```diff
@@ -65,8 +66,8 @@ def evalFunctionsSens(self, funcsSens, config):
             for i in range(self.nCon):
                 p1b, p2b = geo_utils.eDist_b(self.coords[2 * i, :], self.coords[2 * i + 1, :])
                 if self.scaled:
-                    p1b /= self.D0[i]
-                    p2b /= self.D0[i]
+                    p1b /= self.D0[i].real
+                    p2b /= self.D0[i].real
```

Comment:
Is this a bug?

Reply:
`self.D0` is the [initial value of the thickness](https://github.com/mdolab/pygeo/blob/f0595685c539a946b3b8c5c64364673d1dc8aa65/pygeo/constraints/thicknessConstraint.py#L31) at the evaluation points.. in which case is this going to have an imaginary component? can the distance be negative?

Reply:
Hmm it's possible that `self.D0` can stay as real, I don't fully remember why I made this specific change, but the changes in this file are for fixing CS. Maybe the main thing to be fixed is to use the CS-safe version of L2 norm when computing the perturbed distances.

In any case, adding `self.dtype` seems to have broken some tests, so I'll clean up this stuff soon.

Reply:
Yeah I also thought about `self.dtype` as a reason to cast `self.D0`. Will check the fixes when pushed